util/verible-format.py

```diff
@@ -0,0 +1,196 @@
+    parser.add_argument('--inplace', action='store_true', default=False,
+                        help='format files inplace (overwrite original files)')
+    parser.add_argument('-l', '--allowlist', action='store_true', default=False,
+                        help='process only files from allow list')
+    parser.add_argument('-a', '--all', action='store_true', default=False,
+                        help='process all files in repository (.sv and .svh)')
+    parser.add_argument('-f', '--files', metavar='file', type=str, nargs='+',
+                        help='process provided files')
+    args = parser.parse_args()
+
+    if not args.quiet:
+        logger.addHandler(logging.StreamHandler())
+        logger.setLevel(logging.INFO)
+
+        if args.verbose:
+            logger.setLevel(logging.DEBUG)
+
+    if args.inplace:
+        logger.warning('WARNING: Operating in-place - so make sure to make a backup or '
+                       'run this on an experimental branch')
+
+    if not is_verible_formatter_installed():
+        logger.error('verible-verilog-format either not installed or not visible in PATH')
+        sys.exit(1)
+    else:
+        logger.info('Using Verible formatter version: ' + get_verible_version())
+
+    logger.debug('repo_top: ' + get_repo_top())
+    logger.debug('verible exec: ' + str(get_verible_executable_path()))
+    logger.debug('verible version: ' + get_verible_version())
+    logger.debug('verible args: ' + str(VERIBLE_ARGS))
+
+    repo_top_dir = get_repo_top()
+
+    verible_files = []
+
+    if args.files:
+        for f in args.files:
+            filename_abs = repo_top_dir + '/' + f
+            if os.path.exists(filename_abs):
+                verible_files.append((f, filename_abs))
+
+    if args.allowlist:
+        with open(repo_top_dir + '/util/verible-format-allowlist.txt', 'r') as fp:
+            for line in fp:
+                if line[0] == '#':
+                    continue
+
+                filename = line.strip()
+
+                if len(filename) == 0:
+                    continue
+
+                filename_abs = repo_top_dir + '/' + filename
+                if os.path.exists(filename_abs):
+                    verible_files.append((filename, filename_abs))
+
+    if args.all:
+        for root, dirs, files in os.walk(repo_top_dir):
+            # TODO(ldk): Skip '/hw/vendor' and '/hw/top_*'?
+            for f in files:
+                if not (f.endswith('.sv') or f.endswith('.svh')):
+                    continue
+
+                filename_abs = root + '/' + f
+                if os.path.exists(filename_abs):
+                    filename_relative = filename_abs[len(repo_top_dir) + 1:]
+                    verible_files.append((filename_relative, filename_abs))
+
+    logger.debug('files to format: ' + str(verible_files))
+
+    itr_n = 0
+    ret = 0
+    for f in verible_files:
+        r = process_file(f[1], f[0], inplace=args.inplace,
```

Comment:
This would probably be clearer as `for filename_relative, filename_abs in verible_files:`. But I'd actually suggest just storing the absolute filenames in `verible_files` and then deriving the relative filenames here if you need them. In particular, I'm not sure this script will work properly if I pass absolute filenames with the `--files` argument.

Reply:
Keeping only absolute filenames - KISS
I've also modified `--files` behavior. It appends `repo_top_dir` only when relative path is passed.